examples/the-main-function/the_main_function.bal

```diff
@@ -1,26 +1,49 @@
 import ballerina/io;
 
-// The `main` function that accepts student information and prints out a formatted string.
-// The first parameter `name` is a required parameter, while the second parameter `age` is a defaultable
-// parameter with the default value `18`. The third parameter `year` is also a defaultable parameter.
-// The rest parameter `modules` represents the additional arguments.
-// The `main` function may return an `error` or `()`.
-public function main(string name,
-                     int age = 18,
-                     string year = "Freshman",
-                     string... modules)
-                  returns error? {
+// Person record has details about the person.
+public type Person record {|
+    string name;
+    int? age;
+    string[] favoriteColors;
+    boolean lovesCricket;
+|};
 
+// The `main` function accepts Person information and prints the message 
+// `repeats` number of times while prepending the `messages`.
+// The first parameter `repeats` is a required parameter and there can be 
+// 0 or more `messages`. The parameters `repeats` and `messages` are 
+// provided as command line operands.The included record `person` represents 
+// the options given via the command line.
+// The `main` function may return an `error` or `()`.
+public function main(int repeats, string[] messages, *Person person) 
```

Comment:
Some example suggestion


main(string subjectName, string[] studentNames, int[] marks, *ResultOption resultOption)


public type ResultOption record {|
    int passMark;
    boolean? printAverage;	
|};


bal runa main.bal -- --passMArk 50 --printAverage -- Science Tom Alice John 70 30 80

output

Tom    P
Alice  F
John   P 

Average 60

Reply:
With the new command-line args model, you can have at most one operand parameter of type T[]. This example is not valid. Am I missing something? :)

Reply:
According to spec we cannot have `string[]` and `string?` or `string v = "d"`, but it allows having `string`
Quoting from the spec:

> if there is an operand parameter of type O[], then there must be no operand parameters of type O? and O x = d

Reply:
ATM multiple operand parameters with type T[] also allowed and works fine. @riyafa seems spec deviation and needs a fix.

Reply:
@sameerajayasoma I think spec allows having `string[]` and `int[]` arrays. But does not allow having two `string[]` arrays

Reply:
AFAIU we can not have int[] and string[] which are subtypes of basic types and cannot have multiple operand parameters of type T[].

Reply:
@sameerajayasoma can you please clarify, if the spec does or does not allow arrays of different types? I have been under the impression, it does not allow arrays of same type.